In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


The goal of this workbook is to see if there is a statistical sigificance in rideshare rides from yankee stadium on gamedays depending on whether the yankees won or lost.

In [96]:
rideshares = pd.read_parquet(r'C:\Users\EvanSpiller\Documents\DS BootCamp\Capstone\data\HVFHV_Compressed\Combined\full_compressed_HVFHV_2022_2023')

ArrowMemoryError: malloc of size 2544674112 failed

In [98]:
baseball_game = rideshares.loc[
    (
        (
            (rideshares['PULocationID'] == 247)
        )
    & (rideshares['Year'] == 2022)
     & (rideshares['Hour']>=21 )
    )
]


In [99]:
baseball_game

PULocationID  DOLocationID  Year  Month  Day  Hour  trip_miles_sum  \
5570819            247             1  2022      1   17    23          24.150   
5570837            247             3  2022      1    1    22           6.050   
5570838            247             3  2022      1    1    23           6.290   
5570842            247             3  2022      1    2    22           5.880   
5570848            247             3  2022      1    3    23           6.610   
...                ...           ...   ...    ...  ...   ...             ...   
78732395           247           265  2022     12   30    22         103.418   
78732396           247           265  2022     12   30    23          51.740   
78732417           247           265  2022     12   31    21          75.690   
78732418           247           265  2022     12   31    22          36.913   
78732419           247           265  2022     12   31    23          20.820   

          congestion_surcharge_sum  tips_sum  driver_pay_sum  trips  
5570819                        0.0      0.00           56.88      1  
5570837                        0.0      0.00           15.44      1  
5570838                        0.0      0.00           15.84      1  
5570842                        0.0      0.00           16.58      1  
5570848                        0.0      0.00           16.99      1  
...                            ...       ...             ...    ...  
78732395                       0.0      6.27          301.43      3  
78732396                       0.0      0.00          114.60      3  
78732417                       0.0      0.00          199.73      4  
78732418                       0.0     15.75          126.31      4  
78732419                       0.0      3.67           55.08      2  

[59354 rows x 11 columns]

In [100]:
yankees_games = pd.read_csv(r'C:\Users\EvanSpiller\Documents\DS BootCamp\Capstone\data\Yankees_Games_2022.csv')

In [101]:
yankees_games

Date W/L
0      Friday Apr 8   W
1    Saturday Apr 9   W
2     Sunday Apr 10   L
3     Monday Apr 11   L
4    Tuesday Apr 12   W
..              ...  ..
76  Saturday Sep 24   W
77    Sunday Sep 25   W
78    Friday Sep 30   L
79   Saturday Oct 1   W
80     Sunday Oct 2   L

[81 rows x 2 columns]

In [102]:
yankees_games['Date_Proper'] = pd.to_datetime(yankees_games['Date'] + ', 2022', format="%A %b %d, %Y")

In [103]:
yankees_games

Date W/L Date_Proper
0      Friday Apr 8   W  2022-04-08
1    Saturday Apr 9   W  2022-04-09
2     Sunday Apr 10   L  2022-04-10
3     Monday Apr 11   L  2022-04-11
4    Tuesday Apr 12   W  2022-04-12
..              ...  ..         ...
76  Saturday Sep 24   W  2022-09-24
77    Sunday Sep 25   W  2022-09-25
78    Friday Sep 30   L  2022-09-30
79   Saturday Oct 1   W  2022-10-01
80     Sunday Oct 2   L  2022-10-02

[81 rows x 3 columns]

In [104]:
yankees_games.drop('Date', inplace = True, axis = 1)

In [105]:
yankees_games['Date'] = yankees_games['Date_Proper']

In [106]:
baseball_game['Date'] = pd.to_datetime(baseball_game[['Year', 'Month', 'Day']])


C:\Users\EvanSpiller\AppData\Local\Temp\ipykernel_27364\3040419994.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  baseball_game['Date'] = pd.to_datetime(baseball_game[['Year', 'Month', 'Day']])


In [107]:
baseball_game = baseball_game.groupby('Date').sum()['trips']

In [108]:
baseball_game = pd.merge(baseball_game, yankees_games, on='Date', how='inner')

In [109]:
baseball_game = pd.get_dummies(baseball_game, columns=['W/L'])


In [110]:
selected_data = baseball_game.loc[:, ['trips', 'W/L_W']]

In [111]:
baseball_game.rename(columns = {'W/L_W': "Win"}, inplace = True)

In [112]:
baseball_game.drop(['Date_Proper','W/L_L','Date'], axis =1, inplace = True)

In [113]:
Wins = baseball_game.loc[baseball_game['Win']==True]

In [114]:
Losses = baseball_game.loc[baseball_game['Win']==False]

In [115]:
Losses

trips    Win
2    1030  False
3     583  False
5     788  False
14    560  False
19    795  False
20    795  False
21    740  False
37    930  False
38    563  False
43    940  False
45    860  False
46    839  False
52    581  False
54    851  False
55    347  False
56    759  False
57    849  False
59    910  False
60   1045  False
61    531  False
68    736  False
69    757  False
78    916  False
80    359  False

In [116]:
Wins

trips   Win
0     594  True
1     814  True
4     704  True
6     785  True
7     935  True
8     526  True
9     429  True
10    615  True
11    730  True
12    357  True
13    560  True
15    357  True
16    916  True
17    343  True
18    506  True
22    802  True
23    943  True
24    844  True
25    904  True
26    904  True
27    916  True
28    507  True
29    405  True
30    796  True
31    996  True
32    365  True
33    845  True
34    852  True
35   1027  True
36    925  True
39    488  True
40    731  True
41    861  True
42    317  True
44    850  True
47   1168  True
48    417  True
49    962  True
50    959  True
51    497  True
53    717  True
58    736  True
62    418  True
63    848  True
64   1023  True
65    289  True
66    647  True
67    647  True
70    452  True
71    364  True
72    732  True
73    882  True
74    843  True
75   1011  True
76    530  True
77    786  True
79    442  True

In [117]:
from scipy import stats

two_sample_test = stats.ttest_ind(Wins['trips'], Losses['trips']) 

print(two_sample_test)

Ttest_indResult(statistic=-1.0173426641979422, pvalue=0.31209602610873705)


In [118]:
Wins.mean()

trips    698.578947
Win        1.000000
dtype: float64

In [119]:
Losses.mean()

trips    752.666667
Win        0.000000
dtype: float64

.31 p-val but it went down from .56 when I took away away games so I wonder what will happen if I add in 2021 and 2019.